src/components/highlightsSubMenu.svelte

```diff
@@ -0,0 +1,183 @@
+<script lang="ts">
+  import IconButton from '@src/components/iconButton.svelte';
+  import { workStore, saveResumeDataToLocalStorage, type Highlight } from '@src/data/data';
+  import { getTag, addTag, Tag } from '@src/data/tag';
+  import { arrayMove } from '@src/util/arrayMove';
+
+  export let highlights: Highlight[];
+  export let i: number;
+
+  function addHighlight() {
+    workStore.update((workArray) => {
+      workArray[i].highlights.push({
+        visible: true,
+        content: '',
+        tagNames: []
+      });
+      return workArray;
+    });
+  }
+
+  function deleteHighlight(k: number) {
+    if (window.confirm('Are you sure you would like to delete this highlight?')) {
+      workStore.update((workArray) => {
+        workArray[i].highlights.splice(k, 1);
+        return workArray;
+      });
+    }
+  }
+
+  function hideHighlight(k: number) {
+    workStore.update((workArray) => {
+      const visible = workArray[i].highlights[k].visible;
+      workArray[i].highlights[k].visible = !visible;
+      return workArray;
+    });
+  }
+
+  function moveHighlight(k: number, up: boolean) {
+    workStore.update((workArray) => {
+      let highlights = workArray[i].highlights;
+      highlights = up ? arrayMove(highlights, k, k - 1) : arrayMove(highlights, k, k + 1);
+      workArray[i].highlights = highlights;
+      return workArray;
+    });
+  }
+
+  function onTagKeydown(e: KeyboardEvent, k: number) {
+    const target: HTMLInputElement = e.target as HTMLInputElement;
+    const currValue = target ? target.value : '';
+    if (e.key == 'Enter' && currValue.length > 0) {
+      // if the tag is not in the tags, add the tag into the store
+      if (getTag(currValue) == undefined) {
+        addTag(new Tag(currValue));
+      }
+
+      // add tag to the highlight
+      workStore.update((workArray) => {
+        let tagNames = workArray[i].highlights[k].tagNames;
+        if (tagNames == undefined) tagNames = [];
+        tagNames.push(currValue);
+        workArray[i].highlights[k].tagNames = tagNames;
+        return workArray;
+      });
+
+      target.value = '';
+    }
+  }
+
+  /**
+   *
+   * @param e
+   * @param s
+   * @param k the index of the highlight that is being edited
+   * @param saveData
+   */
```

Comment:
This block comment doesn't do much and should be removed. Also should move this function to the highlight related functions for ease of reading